main.go

```diff
@@ -160,3 +166,30 @@ func backwardsCompatibleMetricFactory(configKey string) func(xmetrics.MetricsIn)
 		return registry, nil
 	}
 }
+
+type OnErrorHTTPResponseOption struct {
+	AuthType string
+}
+
+func (o OnErrorHTTPResponseOption) GetAuthType() string {
+	if o.AuthType == "" {
+		return string(basculehttp.BearerAuthorization)
+	}
+	return o.AuthType
+}
+
+func onErrorHTTPResponse(config OnErrorHTTPResponseOption) (basculehttp.OnErrorHTTPResponse, error) {
+	authType := config.GetAuthType()
+	if authType != "Bearer" && authType != "Basic" {
+		return nil, fmt.Errorf("Value not supported: '%s'. Available options are 'Bearer' and 'Basic'.", authType)
```

Comment:
```suggestion
		return nil, fmt.Errorf("value not supported: '%s'. Available options are 'Bearer' and 'Basic'", authType)
```

Overall, this error feels like a log message instead of an error.  I fixed some basic things to make it more error-like, but maybe something like `invalid auth type: expected Basic or Bearer` would work better?